### ISSUE TITLE
SD3 inpaint pipeline can not use EulerA scheduler. bug: The `scale_model_input` function should be called before `step` to ensure correct denoising. 

### ISSUE THREAD

**Issue by huangjun12** [huggingface/diffusers#9084]
### Describe the bug

SD3 inpaint pipeline can not use EulerA scheduler. 

### Reproduction

```python
pipe = StableDiffusion3InpaintPipeline.from_pretrained(xxx)
pipe.scheduler = EulerAncestralDiscreteScheduler.from_config(pipe.scheduler.config)

# then run the pipe
  out_images = pipe(
      prompt=caption, 
      image=input_image, 
      mask_image=input_mask, ).images
```

### Logs

```shell
The `scale_model_input` function should be called before `step` to ensure correct denoising.  See `StableDiffusionPipeline` for a usage example.
```

### System Info

torch: 2.0.0 + cu117
diffusers: commit_id:  `41b705f42db6ce43bbcb41bd2cefd479d8b73774`

**Comment by a-r-r-o-w** [comment]
cc @yiyixuxu There seems to be a line missing after preparing `latent_model_input`:

```
latent_model_input = self.scheduler.scale_model_input(latent_model_input, t)
```

Is this intentionally not present, or was it forgotten? If forgotten, would be awesome if you could open a PR @huangjun12!

**Comment by yiyixuxu** [comment]
our flow match pipelines are meant to work with flow match schedulers, though

**Comment by huangjun12** [comment]
@yiyixuxu Will SD3_inpaint_pipeline support the EulerA scheduler? If Yes, when will it be supported?

**Comment by huangjun12** [comment]
> cc @yiyixuxu There seems to be a line missing after preparing `latent_model_input`:
> 
> https://github.com/huggingface/diffusers/blob/b7058d142c162bc33f1f47f7909e0913ead3d248/src/diffusers/pipelines/stable_diffusion_3/pipeline_stable_diffusion_3.py#L873
> 
> ```
> latent_model_input = self.scheduler.scale_model_input(latent_model_input, t)
> ```
> 
> Is this intentionally not present, or was it forgotten? If forgotten, would be awesome if you could open a PR @huangjun12!

I tried this, but it doesn't seem to work.

**Comment by sauravp** [comment]
@huangjun12 I want to reproduce this. What pre-trained model are you using?
> pipe = StableDiffusion3InpaintPipeline.from_pretrained(xxx)

**Comment by huangjun12** [comment]
> @huangjun12 I want to reproduce this. What pre-trained model are you using?
> 
> > pipe = StableDiffusion3InpaintPipeline.from_pretrained(xxx)

i trained this myself
